src/main/java/org/example/persistence/PersistenceProgram.java

```diff
@@ -1,54 +1,422 @@
+            return resultSet.getInt("language_id");
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return 0;
+        }
+
+    }
 
     @Override
     public boolean createProgram(ProgramEntity programEntity) {
+        long programId = insertProgram();
+        programInformation(programEntity.getDescription(),programEntity.getName(),1,programId);
+        companyProgram(programId,programEntity.getCompany().getId());
+        insertProducer(programEntity,programId);
+        insertCredit(programEntity,programId);
+
         return false;
     }
+    //soft deleting instead of just outright deleting
+    /*public boolean deleteCredit(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement = connection.prepareStatement(
+                    "DELETE FROM credit WHERE program =?");
+
+            statement.setLong(1,programEntity.getId());
+
+            return statement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }*/
+
+    /*public boolean deletecompanyprogram(ProgramEntity programEntity) {
+        try {
+            PreparedStatement  statement = connection.prepareStatement("DELETE FROM companyprogram  " +
+                    "WHERE companyprogram.program = ?");
+
+            statement.setLong(1,programEntity.getId());
+            return statement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }*/
+
+    /* public boolean deleteprogramproducer(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement =connection.prepareStatement("DELETE FROM programproducer where program_id = ? ");
+            statement.setLong(1,programEntity.getId());
+            return  statement.execute();
+        }catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }*/
+
+    /*public boolean deleteProgramTable(ProgramEntity programEntity){
+
+        try {
+            PreparedStatement statement = connection.prepareStatement("DELETE program, programinformation FROM program INNER JOIN " +
+                    "programinformation where program.id = ? and programinformation.id = program.programinformation");
+            statement.setLong(1,programEntity.getId());
+            return statement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+    }*/
+
+
+    public boolean softDeleteProgramTable(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement =connection.prepareStatement("UPDATE program SET timestamp_for_deletion = ? " +
+                    "WHERE id = ? ");
+
+            statement.setTimestamp(1, new Timestamp(System.currentTimeMillis()));
+            statement.setLong(2,programEntity.getId());
+            statement.executeUpdate();
```

Comment:
Hvorfor beder du om et helt object, når du kun bruger id'et for det? hvad er tanken bag det?